src/behavior.scm

```diff
@@ -758,38 +838,92 @@
         ))
 		(TrueLink)
 	))
+;;
+(define (pred-2-schema pnode-str) 
+	(DefineLink 
+		(DefinedSchemaNode pnode-str)
+		(ExecutionOutputLink (GroundedSchemaNode "scm: cog-evaluate!")
```

Comment:
This is fatally flawed: `cog-evaluate!` returns a truth value, but `GroundedSchemaNode` is expecting a function that returns an atom.  I'm surprised that this is not crashing or failing out with an error.

Of course, you could create a new scheme function that returns an atom, here, but what atom should be returned?  I started to invent a new link type -- lets say, ExecLink, that could wrap a predicate and turn it into a schema, but faced exactly that question: what atom should be returned? 

So: this raises a question about OpenPsi: the current OpenPsi seems to want actions to be atom-valued .. but what should the value of those actions be?  i.e. what atom?  Perhaps the right fix is to make OpenPsi use predicates, not schemas, for the action: that way, the TV can be used to indicate if the action succeeded or not.  

So .. anyway: this raises a design issue with OpenPsi that should be answered.


To solve this r